File: src/main/java/no/capraconsulting/siren/internal/util/ListUtil.java
package no.capraconsulting.siren.internal.util;

import java.util.List;
import java.util.function.Function;
import java.util.stream.Collectors;

public final class ListUtil {

    public static <T, R> List<R> map(List<T> list, Function<T, R> map) {
        return list.stream().map(map).collect(Collectors.toList());
    }
}

File: src/main/java/no/capraconsulting/siren/internal/util/ListUtil.kt
@file:JvmName("ListUtil")
package no.capraconsulting.siren.internal.util

import java.util.function.Function

@Deprecated("Use Kotlin stdlib", ReplaceWith("list.map(map::apply)"))
internal fun <T, R> map(list: List<T>, map: Function<T, R>): List<R> =
    list.map(map::apply)
